horizon/facts/update_subscriber.py

```diff
@@ -25,12 +25,15 @@ def _inject_subscriber(self):
     def decorator(self, func):
         @wraps(func)
         async def wrapper(report: DataUpdateReport, *args, **kwargs):
-            await self._resolve_listeners(report.update_id)
+            if report.update_id is not None:
+                self._resolve_listeners(report.update_id)
+            else:
+                logger.debug("Received update report without update ID")
```

Comment:
Not sure if I would call this before or after the regular callbacks 🤔

Reply:
For performance sake, before is better IMO